ingestor-default/src/main/scala/za/co/absa/hyperdrive/ingestor/implementation/decoder/avro/confluent/ConfluentAvroKafkaStreamDecoder.scala

```diff
@@ -55,8 +55,9 @@ private[decoder] class ConfluentAvroKafkaStreamDecoder(val topic: String, val sc
     val schemaRegistryFullSettings = schemaRegistrySettings + (SchemaManager.PARAM_SCHEMA_REGISTRY_TOPIC -> topic)
     logger.info(s"SchemaRegistry settings: $schemaRegistryFullSettings")
 
-    import za.co.absa.abris.avro.AvroSerDe._
-    streamReader.fromConfluentAvro(column = "value", None, Some(schemaRegistryFullSettings))(retentionPolicy)
+    import za.co.absa.abris.avro.functions.from_confluent_avro
+    import org.apache.spark.sql.functions.col
+    streamReader.load().select(from_confluent_avro(col("value"), schemaRegistryFullSettings) as 'data).select("data.*")
```

Comment:
What if `schemaRegistrySettings` is something like `Map[String,String]("" -> "")`? Do we care?

Plus: For your tests, you can leverage the sparkTestBase that gets you a spark session, etc. Can be found in Enceladus. (We should maybe move it to commons)

Reply:
**What if schemaRegistrySettings is something like Map[String,String]("" -> "")? Do we care?**

Hyperdrive itself does not, but ABRiS does, so, if any required configuration is missing, the whole job will be stopped by ABRiS telling exactly which properties are missing.

**Plus: For your tests, you can leverage the sparkTestBase that gets you a spark session, etc. Can be found in Enceladus. (We should maybe move it to commons)**

Great, will have a look. Tks!